realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -542,17 +553,24 @@ public void onChange(RealmResults <Permission> loadedPermissions) {
                 });
             }
         }
+
+        private void notifyCallbackWithSuccess(RealmResults<Permission> permissions) {
+            callback.onSuccess(permissions);
+            activeTasks.remove(this);
```

Comment:
it would be good we can abstract this into the parent, at least the `activeTasks.remove(this);` is the common thing. But I am OK with what it is right now.